### DISCUSSION TITLE
Parametrische Geometrie

### DISCUSSION THREAD

**Discussion by alexrelux** [globallightingdata/l3d#10] [Team Posts]
Hallo, sollen wir für die parametrische Geometrie ein separates Repository machen?

**Comment by robertrelux** [comment]
Hi Alex,

gerne … ich dachte, wir da schon ein Bereich auf GitHub …

Robert

 

Von: alexrelux <notifications@github.com> 
Gesendet: Freitag, 15. Januar 2021 16:58
An: Dx-Rx/dial-relux <DIAL-RELUX@noreply.github.com>
Cc: Subscribed <subscribed@noreply.github.com>
Betreff: [Dx-Rx/dial-relux] Parametrische Geometrie (#2)

 

Hallo, sollen wir für die parametrische Geometrie ein separates Repository machen?

**Comment by Kurpanik** [comment]
Guten Morgen, 

also ich hätte es jetzt eher im vorhandenen LDF Repisitory gesehen, da die Ergebnisse beider XSDs in das Leuchtendaten-XML kommen. Anders als das XML des GDF-XSD, für das es ein eigenes Toolset geben wird. Und daher die Trennung schon eher Sinn macht.

Ein Grund, der noch mehr dafür spricht: Wir wollen das parametrische Geometrie-XML ja nicht als `File` verlinken, sondern in das LDF-XML integrieren (oder war das noch zu klären?). Und dazu müssen sich die XSDs referenzieren. Hier Versionen über zwei Repositories hinweg zu synchronisieren würde eher hinderlich sein.

Macht es vielleicht sogar Sinn, das parametrische XSD in das LdfByseTypes.xsd zu integrieren?

Nachtrag: Eben geschaut, wir haben in einem Meeting schonmal drüber gesprochen und wollten es ins XSD integrieren: https://github.com/Dx-Rx/LDF/issues/31#issuecomment-704145323

**Comment by robertrelux** [comment]
Moin,

richtig, ein eigenes Repo für die parametrische Geometrie macht Sinn. Das mit der Geometire zu kreuzen ist keine gute Idee.

Wir müssen noch klären, ob die XML-Struktur in die LDF Struktur mit rein soll. Ich wäre eher für ein separates File. Das dürfte das Handling und die Verwaltung im PIM vereinfachen. Wir könnten auch Editoren bauen, die losgelost vom LDF parametrische Geometrien erstellen können. Und mehrere Leuchten haben bestimmt die gleichen parametrischen Geometrien (wie bei m3d / r3d auch).

Danke.

 

Von: Alex <notifications@github.com> 
Gesendet: Montag, 18. Januar 2021 08:37
An: Dx-Rx/dial-relux <DIAL-RELUX@noreply.github.com>
Cc: Robert Heinze <r.heinze@relux.com>; Comment <comment@noreply.github.com>
Betreff: Re: [Dx-Rx/dial-relux] Parametrische Geometrie (#2)

 

Guten Morgen,

also ich hätte es jetzt eher im vorhandenen LDF Repisitory gesehen, da die Ergebnisse beider XSDs in das Leuchtendaten-XML kommen. Anders als das XML des GDF-XSD, für das es ein eigenes Toolset geben wird. Und daher die Trennung schon eher Sinn macht.

Ein Grund, der noch mehr dafür spricht: Wir wollen das parametrische Geometrie-XML ja nicht als File verlinken, sondern in das LDF-XML integrieren (oder war das noch zu klären?). Und dazu müssen sich die XSDs referenzieren. Hier Versionen über zwei Repositories hinweg zu synchronisieren würde eher hinderlich sein.

Macht es vielleicht sogar Sinn, das parametrische XSD in das LdfByseTypes.xsd zu integrieren?

**Comment by Kurpanik** [comment]
> richtig, ein eigenes Repo für die parametrische Geometrie macht Sinn.

Falls es kein Tippfehler war: genau das meinte ich oben nicht ;-). Da beides zum Leuchtendaten-XML gehört und die beiden XSDs aufeinander abgestimmt sein müssen. Daher war mein Vorschlag:

1. Repository: LDF - Diese XSDs benötigt man für die Leuchtendaten XML:
  - LdfBaseTypes.xsd
  - **_GeometryTypes.xsd_** (falls getrennt, Name nur Vorschlag)
  - SimpleSchema.xsd
  - ComplexSchema.xsd
2. Repository: GDF - Diese XSDs benötigt man, wenn man zusätzlich 3D Modelle erstellen möchte. Und auch eher intern von unseren Tools, außer Hersteller trauen sich an eigene 3D-Tools
  - Geometry.xsd

> Wir müssen noch klären, ob die XML-Struktur in die LDF Struktur mit rein soll. Ich wäre eher für ein separates File. Das dürfte das Handling und die Verwaltung im PIM vereinfachen. Wir könnten auch Editoren bauen, die losgelost vom LDF parametrische Geometrien erstellen können. Und mehrere Leuchten haben bestimmt die gleichen parametrischen Geometrien (wie bei m3d / r3d auch).

Wir könnten ja beides anbieten? Für simple Leuchten mit einem Quader dürfte das parametrische Model nur einige wenige zusätzliche XML-Zeilen ausmachen. Für die der Overhead einer zusätzliche XML-Datei dann eher hinderlich ist, wenn man sie "mal eben" erzeugen möchte. Oder ohne Container arbeitet, indem man die LVK im XML ebenfalls embedded und einfach nur eine XML haben möchte, die alles beinhaltet, valide ist und in DIALux/Relux importiert werden kann.

**Comment by bschmahl-dx** [comment]
Ich sehe hier auch eher die Integration in das LDF Repo, statt dafür ein neues Repo anzulegen. Aus den bereits genannten Gründen von Alex K.

Weiter bin ich auch für die Integration der parametrischen Geometrie in das LDF XML. Das vereinfacht das Format und es muss nicht noch zusätzlich eine weitere Datei mitgenommen werden.

**Comment by robertrelux** [comment]
Ok, wir können die parametrische Geometrie gerne mit in das LDF Repo legen. 

Ich habe auch nichts dagegen, wenn wir die parametrische Geometrie XSD in die LDF mit einbetten. 

Aber ich stelle mir die Tool Frage. Wenn alles im PIM Enthalten ist, kein Problem. Bis die PIM Systeme Editoren für die parametrische Geometrie haben, wird es etwas dauern.

Wenn aber jemand die parametrische Geometrie mit einem kleinen Tool baut, muss er die Geometrie erst noch zum PIM oder zum LDF Generator bringen (-> XML File).

Es wird auch passieren, dass jemand eine parametrische Geometrie auf viele LDF verteilt. Auch dann wäre ein File nützlich. 

 

 

Von: bschmahl-dx <notifications@github.com> 
Gesendet: Dienstag, 19. Januar 2021 11:26
An: Dx-Rx/dial-relux <DIAL-RELUX@noreply.github.com>
Cc: Robert Heinze <r.heinze@relux.com>; Comment <comment@noreply.github.com>
Betreff: Re: [Dx-Rx/dial-relux] Parametrische Geometrie (#2)

 

Ich sehe hier auch eher die Integration in das LDF Repo, statt dafür ein neues Repo anzulegen. Aus den bereits genannten Gründen von Alex K.

Weiter bin ich auch für die Integration der parametrischen Geometrie in das LDF XML. Das vereinfacht das Format und es muss nicht noch zusätzlich eine weitere Datei mitgenommen werden.

**Comment by alexrelux** [comment]
Das XSD wird recht umfangreich werden. Ich empfehle eine separate Datei.

**Comment by bschmahl-dx** [comment]
Danke für die Infos. Bevor es hier aufgrund des Repos nicht weitergeht, legt doch ein eigenes Repo für die Erstellung an. 

Wenn das XSD der parametrischen Geometrie, dann in einer Version 1 steht, können wir gemeinsam noch mal bewerten, ob wir eine Integration in das LDF vornehmen oder die parametrische Geometrie in einem eigenen Repo bleibt. 

Alle einverstanden?

**Comment by Kurpanik** [comment]
Eine getrennte Datei ist ja absolut ok. Aber dafür ein eigenes Repo pflegen, obwohl sie im anderen XSD referenziert werden muss? Das wird nur zu Unübersichtlichkeit führen und ist umständlich beim entwickeln

**Comment by alexrelux** [comment]
Ich werde die Datei in das bestehende Repo aufnehmen. Bin allerdings noch nicht so weit.

**Comment by bschmahl-dx** [comment]
Super :)

**Comment by Kurpanik** [comment]
> Aber ich stelle mir die Tool Frage. Wenn alles im PIM Enthalten ist, kein Problem. Bis die PIM Systeme Editoren für die parametrische Geometrie haben, wird es etwas dauern.

Daher der Vorschlag weiter oben, beides anzubieten. Ein bißchen vergleichbar, wie mit eingebetteten oder referenzierbaren Dateien. Dann kann der Hersteller entscheiden, was ihm mehr liegt und was sein Tooling hergibt: Enweder er integriert das parametrische Modell gleich mit ins XML. Vielleicht weil er für den Einstieg ohnehin erst nur mit 10 Leuchten startet und das XML von Hand schreibt. Oder er hat ein komplexes System, bei dem er mit Referenzen auf Dateien mit dem Modell besser arbeiten kann.
